apps/frontend/src/pages/quiz/libs/components/quiz-form/libs/helpers/extract-category-ids-from-questions.helper.ts

```diff
@@ -0,0 +1,17 @@
+import { type QuizQuestionDto } from "~/modules/quiz/quiz.js";
+
+function extractCategoryIdsFromQuestions(
+	questions: QuizQuestionDto[][],
+): number[] {
+	const categoryIds = new Set<number>();
+
+	for (const questionsInsideCategory of questions) {
+		for (const question of questionsInsideCategory) {
```

Comment:
Since questions are categorized, you don't have to go through all questions, second loop is redundant